JASP-Desktop/asyncloader.cpp

```diff
@@ -72,6 +72,18 @@ void AsyncLoader::saveTask(const QString &filename, DataSetPackage *package)
 	QString tempFilename = filename + tq(".tmp");
 	try
 	{
+		int maxSleepTime = 2000;
+		int sleepTime = 100;
+		int delay = 0;
+		while (package->analysesHTML.empty())
+		{
+			if (delay > maxSleepTime)
+				break;
+
+			sleep(sleepTime);
+			delay += sleepTime;
+		}
```

Comment:
is this a race condition? what if what you're waiting for takes more than 2 seconds?